tests/Integration/DataManagementSystem/Test_UserMetadata.py

```diff
@@ -0,0 +1,138 @@
+"""
+Test of multi-VO user metadata handling. Assumes a running Dirac instance with the (master?) FileCatalog
+"""
+from __future__ import division
+
+import unittest
+import os
+import sys
+import os.path
+
+from DIRAC.Core.Base.Script import parseCommandLine
+
+parseCommandLine()
+
+from DIRAC.Interfaces.API.Dirac import Dirac
+from DIRAC.Resources.Catalog.FileCatalog import FileCatalog
+
+
+def random_dd(outfile, size_mb):
+  import os
+  with open(outfile, 'w') as f:
+    for i in range(int(size_mb * 2**20 / 512)):
+      f.write(os.urandom(512))
+
+
+class TestUserMetadataTestCase(unittest.TestCase):
+  def setUp(self):
+    self.dirac = Dirac()
+    self.fc = FileCatalog()
+
+    self.lfn5 = '/gridpp/user/m/martynia/FC_test/test_file_10MB_v5.bin'
```

Comment:
This would also mean the tester has to be in the gridpp VO.

Have a look at this file for testing:
https://github.com/DIRACGrid/DIRAC/blob/integration/tests/Integration/DataManagementSystem/Test_DataManager.py
You can adapt your tests so that they are run in the CI.

Reply:
Quite possibly you can add a second FileCatalog instance in the CI that uses MultiVO metadata.

Reply:
I'm looking into this. It's not obvious to me how to add and configure another Catalog. It needs to have FileMetada and DirectoryMetadata options set to multi VO implementations. I sent you an email about it.

Reply:
I answered your email a couple hours ago (unless there was another email?)

In the test: One could add a second filecatalog which uses the same databases as the existing one.

The installation and CS configuration can be done similar to what I have done here:
https://github.com/DIRACGrid/DIRAC/pull/4436/files

Reply:
Thanks. How do youe test the test code ? I found this article: https://dirac.readthedocs.io/en/latest/DeveloperGuide/CodeTesting/ which talks about running  integration tests. Not use if the article is of any use to me. One would ideally want to run the tests in Github. The problem seems to be, however that if I fork off  this branch (as I did) and develop my code intependently, the tests would not run on push, would they? They seem to only be triggered by a pull request, as far as I can see.

Reply:
Well, as this PR is open if you push to the branch for the pull request, the tests will run.

And yes, on that page under https://dirac.readthedocs.io/en/latest/DeveloperGuide/CodeTesting/#running-integration-tests-locally , but you need a very recent version of docker to run this.

Reply:
OK, I followed the manual:

docker run --rm -  ....
It was running fine, until:

Inspecting DIRAC module
2020-03-11 14:44:46 UTC dirac-install [NOTICE]  Installing DIRAC OS master...
Traceback (most recent call last):
  File "/home/dirac/ServerInstallDIR/dirac-install.py", line 2604, in <module>
    if not installDiracOS(releaseConfig):
  File "/home/dirac/ServerInstallDIR/dirac-install.py", line 2305, in installDiracOS
    diracos, diracOSVersion = releaseConfig.getDiracOSVersion(cliParams.diracOSVersion)
AttributeError: 'NoneType' object has no attribute 'getDiracOSVersion'
+ echo 'ERROR: dirac-install.py failed'
+ exit 1
ERROR: dirac-install.py failed

The OS is, if it matters:

[centos@janusz-rucio DIRAC]$ uname -a
Linux janusz-host1.novalocal 3.10.0-957.27.2.el7.x86_64 #1 SMP Mon Jul 29 17:46:05 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

Reply:
Maybe there is a hint in the rest of the output? OS shouldn't matter as that error should be inside docker... maybe @chrisburr has an idea?

Reply:
That means the `releases.cfg` file wasn't found (`dirac-install` should really print a more helpful message). You should look higher up in the output for which URL it tried to download.

Reply:
There was a hint, indeed. I was on a wrong branch. It looks better now, still installing various components and running tests.

Reply:
The error reappered again this afternoon. I succesfully ran the docker installation this morning (I mean dirac components were installed, including my new FileCatalog service). In a process of testing the clients I restarted the installation and it failed again). From what I can see it reads:
/home/dirac/ServerInstallDIR/dirac-install.py -ddd --dirac-os --dirac-os-version=master --module=/home/dirac/LocalRepo/ALTERNATIVE_MODULES/DIRAC:::DIRAC:::local /home/dirac/ServerInstallDIR/install.cfg /home/dirac/ServerInstallDIR/dirac-ci-install.cfg

Shouldn't it be installing from my branch ? (remotes/origin/rel-v7r1-mVO_metadata_tests_dev). This is the branch I _cd_ to berore running _docker run_ . 
The branch is listed in the output.
Has anything changed between this morning and ~12pm

Reply:
New pre-releases were tagged today so it will have been an issue until the tarballs were uploaded.

Hopefully the tagging of new releases will be automated soon so this can stop being a problem...